saleor/webhook/observability/worker.py

```diff
@@ -0,0 +1,190 @@
+                        events_count += 1
+                    except Exception:
+                        self._queue.task_done()
+                        logger.error(
+                            "[Observability] Failed generating payload", exc_info=True
+                        )
+                if events_count:
+                    with opentracing_trace("put_events", "buffer"):
+                        try:
+                            buffer_put_multi_key_events(events)
+                        except Exception:
+                            logger.error(
+                                "[Observability] Buffer error, dropped %s events",
+                                events_count,
+                                exc_info=True,
+                            )
+                for _ in range(events_count):
+                    self._queue.task_done()
+        logger.debug("[Observability] Background worker stopped")
+
+    def start(self):
+        with self._lock:
+            if not self.is_alive:
+                self._thread = Thread(
+                    target=self._target,
+                    name="observability_worker",
+                    daemon=True,
+                )
+                self._thread.start()
+                self._thread_for_pid = os.getpid()
+
+    @property
+    def is_alive(self) -> bool:
+        if self._thread_for_pid != os.getpid() or not self._thread:
+            return False
+        return self._thread.is_alive()
+
+    def _ensure_thread(self):
+        if not self.is_alive:
+            self.start()
+
+    def submit_event(
+        self, buffer_name: str, generate_payload: Callable[[], Any]
+    ) -> bool:
+        self._ensure_thread()
+        try:
+            self._queue.put_nowait((buffer_name, generate_payload))
+            return True
+        except Full:
+            logger.warning("[Observability] Queue full, event dropped")
+        return False
+
+    def queue_join(self, timeout: float) -> bool:
+        deadline = monotonic() + timeout
+        queue = self._queue
+        with queue.all_tasks_done:
+            while queue.unfinished_tasks:
+                delay = deadline - monotonic()
+                if delay <= 0:
+                    return False
+                queue.all_tasks_done.wait(timeout=delay)
+        return True
+
+    def _wait_flush(self, timeout: float) -> bool:
+        initial_timeout = min(0.1, timeout)
+        if not self.queue_join(initial_timeout):
+            pending = self._queue.qsize()
+            logger.debug("[Observability] %d events pending on flush", pending)
+            if not self.queue_join(timeout - initial_timeout):
+                pending = self._queue.qsize()
+                logger.error(
+                    "[Observability] Flush timed out, dropped %s events", pending
+                )
+                return False
+        return True
+
+    def flush(self, timeout: float) -> bool:
```

Comment:
That name is confusing, I expect the queue to be flushed but the worker thread to remain alive. I would suggest having `flush()` but the shutdown part to be `shutdown()` method

Reply:
I agree. Fixing that 👍